src/Core/Search/Discussion/Gambit/AuthorGambit.php

```diff
@@ -50,7 +51,7 @@ protected function conditions(AbstractSearch $search, array $matches, $negate)
 
         $ids = [];
         foreach ($usernames as $username) {
-            $ids[] = $this->users->getIdForUsername($username);
+            $ids[] = is_numeric($username) ? $username : $this->users->getIdForUsername($username);
```

Comment:
Both solutions are very flaky. Mostly because we tolerate digit usernames as well, doing a intval or filter_var will only assume a username to be a user ID. Wouldn't it make more sense to stick to usernames only for now?